### DISCUSSION TITLE
Clarifications needed on beams

### DISCUSSION THREAD

**Discussion by hbitteur** [Audiveris/audiveris#474] [Q&A]
I would appreciate help on how to understand the measure below (taken from "the man I love", sheet 3):

![image](https://user-images.githubusercontent.com/22053634/111799674-87062e80-88cb-11eb-8266-28ee87933e7b.png)
In the upper left corner, we have two overlapping symbols. Here, we cannot reasonably blame OMR engine for not recognizing musical symbols. This is funny, but can easily be fixed manually.
The problems are in the beam group located at measure center: a long beam with one hook above and one hook below.

1/ Current Audiveris OMR engine considers that time slots along a beam are governed by abscissa of connections between a beam and its connected stems. The configuration above is the first counter-example I ever encountered.
So the engine algorithm needs to be modified. We'll apply at beam-connected chords the same rule used by non-beamed chords, which is to use the middle abscissa of the chord bounds (defined as the bounding envelop of heads and stem).
This will allow the beamed-chords above and below  the red cross to share the same time slot, despite the fact that their stems are not aligned.
Fine.

2/ What appears to be more difficult is the voices assignment. Current OMR engine assumes that all chords connected via the same beam group belong to the same voice. In the measure at hand, we have one beam group. Even with manual adjustments, the best configuration I was able to obtain is as depicted below:
![image](https://user-images.githubusercontent.com/22053634/111802025-d2b9d780-88cd-11eb-82eb-80572df73f33.png)
It looks as if chord 5857 (black head with 2 flags) and following chord 5858 (beamed dotted head) should belong to the same voice. And similarly, chords 5868 & 5869, as well as chords 5859 & 5861 and chords 5870 & 5871.
But in that case, the upper chords of the beam group and the lower chords of the same beam group would belong to separate voices.
Does this make sense? In other words, in which direction should I modify engine algorithm (or leave it as it is)?
Thanks in advance for you help
/Hervé

**Comment by Bacchushlg** [comment]
The correct interpretation of the measure should look like this:

![image](https://user-images.githubusercontent.com/33522630/111866286-aef99e80-896c-11eb-8c79-28c553850bf6.png)

But the formatting of the used scoring program is horrible. Especially concerning the position of the upper 1st rest, which belongs to the lower voice, but also about the beams that overlap (no go!)

**Reply by hbitteur** [reply]
I did not think of this: indeed, we could consider that we have in fact two overlapping beams!
But, never expect the OMR engine to detect this "_auto-magically_"...
